File: ui/common/src/main/groovy/com/github/thetric/iliasdownloader/ui/common/prefs/UserPreferences.java
package com.github.thetric.iliasdownloader.ui.common.prefs;

import lombok.Data;

import java.util.Collection;
import java.util.HashSet;

/**
 * Contains the user preferences.
 *
 * @see UserPreferenceService
 * @see YamlUserPreferenceService
 */
@Data
public final class UserPreferences {
    private String iliasServerURL;

    private String userName;

    /**
     * Limit per download in mebibytes. Zero means unlimited
     */
    private long maxFileSizeInMiB;

    /**
     * Course IDs to include in the sync
     */
    private Collection<Long> activeCourses = new HashSet<>();
}

File: src/main/kotlin/com/github/thetric/iliasdownloader/ui/common/prefs/UserPreferences.kt
package com.github.thetric.iliasdownloader.ui.common.prefs

/**
 * Contains the user preferences.
 *
 * @see PreferenceService
 * @see JsonPreferenceService
 */
data class UserPreferences(
    var iliasServerURL: String,
    val userName: String,

    /**
     * Limit per download in mebibytes. Zero means unlimited
     */
    val maxFileSizeInMiB: Long,

    /**
     * Course IDs to include in the sync
     */
    val activeCourses: Collection<Long> = HashSet()
)
